examples/nlp/duplex_text_normalization/duplex_text_normalization_infer.py

```diff
@@ -60,6 +59,15 @@
 from nemo.core.config import hydra_runner
 from nemo.utils import logging
 
+try:
+    from nemo_text_processing.text_normalization.data_loader_utils import post_process_punct
+except ModuleNotFoundError:
+    raise ModuleNotFoundError(
```

Comment:
Dont raise at runtime. It will not fix anything cause we have multiple `__init__` that imports from these files and so this path will be called. i.e. - even if its unused, import will still raise this error.


Do the delayed check please - 
First create a bool - https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/asr/losses/rnnt.py#L53-L58 
Next check bool at runtime - https://github.com/NVIDIA/NeMo/blob/2cc09425aba3e9b3cfdba43a3188eaef58227055/nemo/collections/asr/losses/rnnt.py#L143-L150

Or at init - https://github.com/NVIDIA/NeMo/blob/2cc09425aba3e9b3cfdba43a3188eaef58227055/nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py#L85-L88

Reply:
You can check this by removing the nemo text processing from the requirements, all these imports will raise error

Reply:
I don't think this is necessary. All the files that are touched in this PR are under either examples, scripts, or tutorials. None of these files should be imported in the NeMo library.

Reply:
But this doesn't actually fix the issue - you still have NeMo toolkit in requirements, so pip install all still can't be done when pynini is not available